recursion/reading-recursive-code.md

```diff
@@ -0,0 +1,288 @@
+| Recursive Case               | A situation in a recursive function that will call the function itself                                                                                                | "When computing the nth fibonacci number, if `n` is greater than 2, the `fibonacci` function will call itself."                                        |
+| Stack (data structure)       | A data structure that operates as "Last-In-First-Out," where new data is pushed onto the top, and data is popped off from the top                                     |
+| System stack                 | A part of memory dedicated to storing the method calls and local variables of a running program. The system stack is responsible for keeping track of function calls. |
+
+## Recursion
+
+In programming, a recursive function is a function that calls itself.
+
+We can imagine a recursive function named `get_next_item`. When we call `get_next_item`, it will invoke itself, `get_next_item`. Of course, when _this_ function calls begins execution, it will also invoke itself.
+
+Both iteration and recursion are programming techniques used to repeat logic and behavior. There are often iterative and recursive solutions to the same problem!
+
+When would we choose to solve a problem with recursion? Once we're used to thinking about recursion, recursive algorithms may be more readable and understandable compared to iterative solutions.
+
+## Anatomy of a Recursive Function
+
+Functions that call itself, like the imaginary `get_next_item` function, will recurse an infinite number of times.
+
+However, if a function calls itself an infinite number of times, when will we actually get an answer and solve our problem? Recursive functions are more helpful when they actually _do_ stop!
+
+In order to better read and debug recursive functions, let's dissect the anatomy of one.
+
+### Base Cases are Stopping Conditions
+
+Recursive algorithms will usually take in at least one argument. A recursive algorithm handles different situations based on the value of the argument. The algorithm should handle:
+
+- At least one _recursive case_
+- At least one _base case_
+
+| Part           | Definition                                                            |
+| -------------- | --------------------------------------------------------------------- |
+| Recursive Case | A case or situation that depends on the solution of other cases       |
+| Base Case      | A case or situation where there is a solution without using recursion |
+
+The responsibility of the _recursive case_ is to divide the problem into a smaller problem.
+
+The responsibility of the _base case_ is to give a solid, immediate solution to the problem.
```

Comment:
swap these as above
reword base case to reinforce that it solves a certain version of the problem
for recursive, prefer transform over divide since not all recursive solutions are divide and conquer, and emphasize that it should be heading for the base case
```suggestion
The responsibility of the _base case_ is to immediately answer a particular form of the problem.

The responsibility of the _recursive case_ is to transform the current problem into one that can be expressed as a "smaller" version of the same problem. Eventually it should reach a configuration that can be handled by the base case.
```